tests/test_history.py

```diff
@@ -10,19 +10,23 @@
 from pyte.compat import map, str
 
 
-def chars(lines):
-    return ["".join(map(operator.attrgetter("data"), line))
-            for line in lines]
+def chars(lines, columns):
```

Comment:
Could you do this with a generator expression? Something along the lines of

```python
["".join(line[column].data for column in range(columns))
 for line in lines]
```

A better name for `lines` would probably be `history_lines` or maybe even just `history`. Feel free to choose what you think is best. I was clearly very bad in naming things back then :)